code/controllers/CommentingController.php

```diff
@@ -207,6 +208,36 @@ public function approve() {
 	}
 	
 	/**
+	 * Renders html for replying to threaded comments
+	 * Should be called via AJAX
+	 * Example of url: /CommentingController/replyform/5?ReturnURL=/blog/sample-blog-entry/
+	 * 
+	 * @return string
+	 */
+	public function replyform(){
+		$p = $this->getURLParams();
+		$cID = (int) $p['ID'];
+		$c = Comment::get()->ByID($cID);
+
+		$controller = new CommentingController();		
+		$controller->setOwnerRecord($c);
+		$controller->setBaseClass('Comment');
+		$controller->setOwnerController(Controller::curr());
+		
+		$form = $controller->CommentsForm();
+		
+		//setting return url
+		if (isset($_GET['ReturnURL'])) {
+			$form->loadDataFrom(array(
+				'ReturnURL'		=> $_GET['ReturnURL'],
+			));
+		}
+		
+		return $form->forTemplate();
```

Comment:
You don't need to explicitly render the form. If you don't then it makes extending the form easier.